src/components/MyAccount/NewSettings/EmailForm.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+  Icon,
+  TextInput,
+  Text,
+  Flex,
+  Button,
+} from "@nypl/design-system-react-components"
+import { useContext, useState } from "react"
+import { PatronDataContext } from "../../../context/PatronDataContext"
+
+const EmailForm = ({ patronData, setIsLoading, setIsSuccess }) => {
+  const { getMostUpdatedSierraAccountData } = useContext(PatronDataContext)
+  const [emails, setEmails] = useState(patronData?.emails || [])
+  const [isEditing, setIsEditing] = useState(false)
+  const [error, setError] = useState(false)
+
+  const [tempEmails, setTempEmails] = useState([...emails])
+
+  const validateEmail = (email) => {
+    const emailRegex = /^[^@]+@[^@]+\.[^@]+$/
+    return emailRegex.test(email)
+  }
+
+  const handleInputChange = (e, index) => {
+    const { value } = e.target
+    const updatedEmails = [...tempEmails]
+    updatedEmails[index] = value
+    setTempEmails(updatedEmails)
+
+    // The first email entry cannot be empty.
+    if (index === 0 && (!value || !validateEmail(value))) {
+      setError(true)
+    } else {
+      const hasInvalidEmail = updatedEmails.some(
+        (email) => !validateEmail(email)
+      )
+      setError(hasInvalidEmail)
+    }
+  }
+
+  const handleRemoveEmail = (index) => {
+    const updatedEmails = tempEmails.filter((_, i) => i !== index)
+    setTempEmails(updatedEmails)
+
+    // Immediately revalidate remaining emails.
+    const hasInvalidEmail = updatedEmails.some((email) => !validateEmail(email))
+    setError(hasInvalidEmail)
+  }
+
+  const handleAddEmail = () => {
+    const updatedEmails = [...tempEmails, ""]
+    setTempEmails(updatedEmails)
+
+    // Immediately revalidate all emails.
+    const hasInvalidEmail = updatedEmails.some((email) => !validateEmail(email))
+    setError(hasInvalidEmail)
+  }
+
+  const handleClearableCallback = (index) => {
+    const updatedEmails = [...tempEmails]
+    updatedEmails[index] = ""
+    setTempEmails(updatedEmails)
+    setError(true)
+  }
+
+  const cancelEditing = () => {
+    setTempEmails([...emails])
+    setIsEditing(false)
+    setError(false)
+  }
+
+  const submitEmails = async () => {
+    setIsLoading(true)
+    const validEmails = tempEmails.filter((email) => validateEmail(email))
+
+    try {
+      const response = await fetch(
+        `/research/research-catalog/api/account/settings/${patronData.id}`,
+        {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ emails: validEmails }),
+        }
+      )
+
+      if (response.status === 200) {
+        await getMostUpdatedSierraAccountData()
+        setIsSuccess(true)
+        setIsEditing(false)
+      } else {
+        setTempEmails([...emails])
+      }
+    } catch (error) {
+      console.error("Error submitting emails:", error)
```

Comment:
Similar to the success banner, if there's an error the user should know something went wrong. The current banner only displays the success message so that will need an update.

This is not in the req/ac for this ticket so it could be separate work.

Reply:
Added a very basic failure banner, "Your changes were not saved"... should I ask Apoorva for copy?

Reply:
Add the basic failure banner with the suggested copy and we'll run it by her during VQA.

Reply:
Where can I find the suggested copy? @EdwinGuzman